### ISSUE TITLE
[QUESTION] Will megatron support norm_topk_prob

### ISSUE THREAD

**Issue by chengfx** [NVIDIA/Megatron-LM#1593]
Qwen3 MoE enables the norm_topk_prob like below in huggingface transformers.
https://github.com/huggingface/transformers/blob/main/src/transformers/models/qwen3_moe/modeling_qwen3_moe.py#L243

```python
if self.norm_topk_prob:  # only diff with mixtral sparse moe block!
            routing_weights /= routing_weights.sum(dim=-1, keepdim=True)
```

I didn't find this feature in Megatron. Will megatron support it?  I only find **scaling_factor** but it is different
https://github.com/NVIDIA/Megatron-LM/blob/main/megatron/core/transformer/moe/moe_utils.py#L569

```python
if scaling_factor:
        probs = probs * scaling_factor
```

**Comment by lostkevin** [comment]
According to [doc](https://docs.pytorch.org/docs/stable/generated/torch.nn.Softmax.html), the following code:
```
        routing_weights = F.softmax(router_logits, dim=1, dtype=torch.float) # sigmoid and normalize on all scores
        routing_weights, selected_experts = torch.topk(routing_weights, self.top_k, dim=-1)
        routing_weights /= routing_weights.sum(dim=-1, keepdim=True) # then renormalize on top-k
```
is equivalent to the follows (set use_pre_softmax=False):
```
        routing_weights_pre_softmax, selected_experts = torch.topk(router_logits, self.top_k, dim=-1)
        routing_weights = torch.softmax(routing_weights_pre_softmax, dim=1) # sigmoid and normalize on top-k scores
```

**Comment by chengfx** [comment]
Got it. Thanks for clarification

**Comment by binxuan** [comment]
> According to [doc](https://docs.pytorch.org/docs/stable/generated/torch.nn.Softmax.html), the following code:
> 
> ```
>         routing_weights = F.softmax(router_logits, dim=1, dtype=torch.float) # sigmoid and normalize on all scores
>         routing_weights, selected_experts = torch.topk(routing_weights, self.top_k, dim=-1)
>         routing_weights /= routing_weights.sum(dim=-1, keepdim=True) # then renormalize on top-k
> ```
> 
> is equivalent to the follows (set use_pre_softmax=False):
> 
> ```
>         routing_weights_pre_softmax, selected_experts = torch.topk(router_logits, self.top_k, dim=-1)
>         routing_weights = torch.softmax(routing_weights_pre_softmax, dim=1) # sigmoid and normalize on top-k scores
> ```

Are these two equivalent for backprop as well? will topk(softmax(router_logits)) backprop to all experts?
